crates/oxc_transformer/src/regexp/mod.rs

```diff
@@ -113,9 +113,10 @@ impl<'a> Traverse<'a> for RegExp<'a> {
         expr: &mut Expression<'a>,
         ctx: &mut oxc_traverse::TraverseCtx<'a>,
     ) {
-        let Expression::RegExpLiteral(ref mut regexp) = expr else {
+        let Expression::RegExpLiteral(regexp) = expr else {
             return;
         };
+        let regexp = &mut **regexp;
```

Comment:
Maybe we can use `DerefMut` here, I always find too many symbols to be confusing compared to this:

```rust
let regexp = regexp.deref_mut();
```

It's just a personal opinion, Though I may as well share it in case you find it tasteful.

Reply:
I agree! `&mut **` is rather confusing. But clippy seems to disagree (always ruining the fun, clippy!)

<img width="507" alt="deref" src="https://github.com/user-attachments/assets/5ce16d34-0ce6-44cb-a5eb-6d0b11836d47">

I think what we need is `Box::as_mut`. #5515

Reply:
It's now `regexp.as_mut()`. Much better.